### PR TITLE
Make editting prebuilt create a copy instead of editting the prebuilt

### PR DESCRIPTION
**Pull request by Magmanat**

### CODE REVIEW

File: src/main/java/seedu/bigpp/pc/PC.java [hunk 1]
```diff
@@ -30,11 +30,25 @@ public PC(String name, Boolean isPrebuilt) {
         this.isPrebuilt = isPrebuilt;
     }
 
+    public PC(PC pcCopy) {
```
Comment (Reynold-SL) [inline src/main/java/seedu/bigpp/pc/PC.java:33]: Code looks good and is easy to follow

File: src/main/java/seedu/bigpp/command/viewercommand/ViewerEditCommand.java [hunk 2]
```diff
@@ -38,6 +39,13 @@ public String executeCommand(DataStorage dataStorage)
             throw new PPIndexOutOfBoundsException();
         }
 
+        // if the PC is pre-built, create a copy of the PC to edit
+        if (dataStorage.pcList.get(pcIndex).getIsPreBuilt()) {
+            PC pcCopy = new PC(dataStorage.pcList.get(pcIndex));
```
Comment (Reynold-SL) [inline src/main/java/seedu/bigpp/command/viewercommand/ViewerEditCommand.java:44]: Code looks good

**Review by Reynold-SL — APPROVED** [review]

File: src/main/java/seedu/bigpp/pc/PC.java [hunk 3]
```diff
@@ -30,11 +30,25 @@ public PC(String name, Boolean isPrebuilt) {
...
+        this.cpuCooler = pcCopy.getCpuCooler();
+        this.gpu = pcCopy.getGpu();
+        this.motherboard = pcCopy.getMotherboard();
+        this.ram = pcCopy.getRam();
+        this.storage = pcCopy.getStorage();
+        this.psu = pcCopy.getPsu();
+        this.chassis = pcCopy.getChassis();
+        this.budget = pcCopy.getBudget();
+    }
+
```
Comment (typingpanda) [inline src/main/java/seedu/bigpp/pc/PC.java:46]: is overloading the constructor a good idea? or would it be better to create a method for creating a pc copy

**Review by typingpanda — APPROVED** [review]
I leave it up to your discretion regarding the PC constructor

File: src/main/java/seedu/bigpp/pc/PC.java (hunk 3 above)
Comment (Magmanat) [reply src/main/java/seedu/bigpp/pc/PC.java:46]: Think it is ok, if it grows into an issue we can change it out

**Pull request merged**